cashapppay/src/main/java/com/adyen/checkout/cashapppay/internal/ui/model/CashAppPayComponentParamsMapper.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 26/6/2023.
+ */
+
+package com.adyen.checkout.cashapppay.internal.ui.model
+
+import com.adyen.checkout.cashapppay.CashAppPayConfiguration
+import com.adyen.checkout.cashapppay.CashAppPayEnvironment
+import com.adyen.checkout.components.core.internal.ui.model.ComponentParams
+import com.adyen.checkout.components.core.internal.ui.model.SessionParams
+import com.adyen.checkout.core.Environment
+
+internal class CashAppPayComponentParamsMapper(
+    private val overrideComponentParams: ComponentParams?,
+    private val overrideSessionParams: SessionParams?,
+) {
+
+    fun mapToParams(
+        configuration: CashAppPayConfiguration,
+        sessionParams: SessionParams?,
+    ): CashAppPayComponentParams = configuration
+        .mapToParamsInternal()
+        .override(overrideComponentParams)
+        .override(sessionParams ?: overrideSessionParams)
+
+    private fun CashAppPayConfiguration.mapToParamsInternal() = CashAppPayComponentParams(
+        isSubmitButtonVisible = isSubmitButtonVisible ?: true,
+        shopperLocale = shopperLocale,
+        environment = environment,
+        clientKey = clientKey,
+        isAnalyticsEnabled = isAnalyticsEnabled ?: true,
+        isCreatedByDropIn = false,
+        amount = amount,
+        cashAppPayEnvironment = cashAppPayEnvironment ?: getDefaultCashAppPayEnvironment(this),
```

Comment:
Nitpick: check out `getGooglePayEnvironment` in `GooglePayComponentParamsMapper` in case you find it nicer